runner/src/elf2tab.rs

```diff
@@ -4,9 +4,32 @@ use std::io::ErrorKind;
 use std::path::PathBuf;
 use std::process::Command;
 
+fn get_platform_architecture(platform: &str) -> Option<&'static str> {
+    match platform {
+        "raspberrypi_pico" | "nano_rp2040_connect" => Some("cortex-m0"),
+        "apollo3"
+        | "clue_nrf52840"
+        | "hail"
+        | "imix"
+        | "microbit_v2"
+        | "msp432"
+        | "nano33bl3"
+        | "nrf52"
+        | "nrf52840"
+        | "nucleo_f429zi"
+        | "nucleo_f446re"
+        | "stm32f3discovery"
+        | "stm32f412gdiscovery" => Some("cortex-m4"),
+        "imxrt1050" | "teensy40" => Some("cortex-m7"),
+        "opentitan" | "esp32_c3_devkitm_1" => Some("riscv32imc"),
+        "hifive1" => Some("riscv32imac"),
+        _ => None,
```

Comment:
Could this be derived from the target triple by default?
As far as I understand, targets with the same target triple will get the same binary from rustc anyway, and deriving it would keep small the number of places to modify when adding a platform.